runelite-client/src/main/java/net/runelite/client/plugins/multicombatborders/MultiCombatBordersPlugin.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.multicombatborders;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.google.inject.Provides;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.opponentinfo.OpponentInfoPlugin;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+@PluginDescriptor(
+	name = "Multicombat borders",
+	description = "Show borders between mutli and single combat areas"
+)
+public class MultiCombatBordersPlugin extends Plugin
+{
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private MultiCombatBordersOverlay multiCombatBordersOverlay;
+
+	List<MultiCombatBorder> multiCombatBorders = loadMultiCombatBorders();
+
+	@Provides
+	MultiCombatBordersConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(MultiCombatBordersConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(multiCombatBordersOverlay);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		overlayManager.remove(multiCombatBordersOverlay);
+	}
+
+	private List<MultiCombatBorder> loadMultiCombatBorders()
+	{
+		Gson gson = new Gson();
+		InputStream multiBordersFile = OpponentInfoPlugin.class.getResourceAsStream("/multi_borders.json");
+		JsonArray items = gson.fromJson(new InputStreamReader(multiBordersFile), JsonArray.class);
```

Comment:
Gson supports deserialization into objects which is generally preferred since it is more readable: https://github.com/google/gson/blob/master/UserGuide.md#TOC-Object-Examples

Reply:
Given the structure of the JSON I don't think using the default object deserialization would work any good. Accessing coordinates would be awkward due to the map where the key changes for every object.

```json
[ 
  {
    "0" : {
      "single" : {
        "x" : 3135,
        "y" : 3523
      },
      "multi" : {
        "x" : 3136,
        "y" : 3523
      }
    }
  },
  {
    "1": { ... }
  }
]
```